src/api/worker/facades/UserFacade.ts

```diff
@@ -83,7 +84,17 @@ export class UserFacade implements AuthHeadersProvider {
 
 	getUserGroupKey(): Aes128Key {
 		// the userGroupKey is always written after the login to this.groupKeys
-		return assertNotNull(this.groupKeys.get(this.getUserGroupId()), "User is not logged in")
+		//if the user has only logged in offline this has not happened
+		const userGroupKey = this.groupKeys.get(this.getUserGroupId())
+		if (userGroupKey == null) {
+			if (this.isPartiallyLoggedIn()) {
+				throw new LoginIncompleteError("userGroupKey not available")
+			} else {
+				throw new Error("User is not logged in")
```

Comment:
`getUserGroupKey` previously asserted not null, why is `userGroupKey === null` possible now even when not partially logged in? is this a case of an "unreachable" situation?

Reply:
It shouldn't be reached, it's equivalent of what would happen before. The only way to be fully logged in is to unlock the key.

Reply:
then I'd propose to make this a ProgrammingError and put that info in the message